extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/vector/CharVectorExpansionKernel.java

```diff
@@ -12,69 +12,119 @@
 
+    private static final String DEBUG_NAME = "CharVectorExpansionKernel";
+
     @Override
     public <A extends Any> WritableChunk<A> expand(
-            final ObjectChunk<Vector<?>, A> source, final WritableIntChunk<ChunkPositions> perElementLengthDest) {
+            @NotNull final ObjectChunk<CharVector, A> source,
+            final int fixedSizeLength,
+            @Nullable final WritableIntChunk<ChunkPositions> offsetsDest) {
         if (source.size() == 0) {
-            perElementLengthDest.setSize(0);
+            if (offsetsDest != null) {
+                offsetsDest.setSize(0);
+            }
             return WritableCharChunk.makeWritableChunk(0);
         }
 
         final ObjectChunk<CharVector, A> typedSource = source.asObjectChunk();
 
         long totalSize = 0;
-        for (int i = 0; i < typedSource.size(); ++i) {
-            final CharVector row = typedSource.get(i);
-            totalSize += row == null ? 0 : row.size();
+        for (int ii = 0; ii < typedSource.size(); ++ii) {
+            final CharVector row = typedSource.get(ii);
+            long rowLen;
+            if (fixedSizeLength != 0) {
+                rowLen = Math.abs(fixedSizeLength);
+            } else {
+                rowLen = row == null ? 0 : row.size();
+            }
+            totalSize += rowLen;
         }
         final WritableCharChunk<A> result = WritableCharChunk.makeWritableChunk(
-                LongSizedDataStructure.intSize("ExpansionKernel", totalSize));
+                LongSizedDataStructure.intSize(DEBUG_NAME, totalSize));
         result.setSize(0);
 
-        perElementLengthDest.setSize(source.size() + 1);
-        for (int i = 0; i < typedSource.size(); ++i) {
-            final CharVector row = typedSource.get(i);
-            perElementLengthDest.set(i, result.size());
-            if (row == null) {
-                continue;
+        if (offsetsDest != null) {
+            offsetsDest.setSize(source.size() + 1);
+        }
+        for (int ii = 0; ii < typedSource.size(); ++ii) {
+            final CharVector row = typedSource.get(ii);
+            if (offsetsDest != null) {
+                offsetsDest.set(ii, result.size());
+            }
+            if (row != null) {
+                final CharConsumer consumer = result::add;
+                try (final CloseablePrimitiveIteratorOfChar iter = row.iterator()) {
+                    Stream<Character> stream = iter.stream();
+                    if (fixedSizeLength > 0) {
+                        // limit length to fixedSizeLength
+                        stream = stream.limit(fixedSizeLength);
+                    } else if (fixedSizeLength < 0) {
+                        final long numToSkip = Math.max(0, row.size() + fixedSizeLength);
+                        if (numToSkip > 0) {
+                            // read from the end of the array when fixedSizeLength is negative
+                            stream = stream.skip(numToSkip);
```

Comment:
Where are we hitting the tail case, I don't see this getting hit in either the vector or array kernels.